compiler/basicTypes/MkId.hs

```diff
@@ -961,7 +961,7 @@ wrapCo :: Coercion -> Type -> (Unboxer, Boxer) -> (Unboxer, Boxer)
 wrapCo co rep_ty (unbox_rep, box_rep)  -- co :: arg_ty ~ rep_ty
   = (unboxer, boxer)
   where
-    unboxer arg_id = do { rep_id <- newLocal (linear rep_ty)
+    unboxer arg_id = do { rep_id <- newLocal (Scaled (idMult' arg_id) rep_ty)
```

Comment:
I'm finding this very curious. It's almost certainly the right thing to do (assuming the `arg_id` correspond to a field, the unboxer must use the same multiplicity as the field). I just fail to see why it affects our test case.